src/cmd.rs

```diff
@@ -77,6 +154,165 @@ pub fn osd_pool_set(cluster_handle: rados_t, pool: &str, key: &str, value: &str,
     Ok(())
 }
 
+pub fn osd_set(cluster_handle: rados_t, key: &str, force: bool, simulate: bool) -> Result<(), RadosError> {
+    let cmd = match force {
+        true => {
+            json!({
+                "prefix": "osd set",
+                "key": key,
+                "sure": "--yes-i-really-mean-it",
+            })
+        },
+        false => {
+            json!({
+                "prefix": "osd set",
+                "key": key,
+            })
+        },
+    };
+    debug!("osd set: {:?}", cmd.to_string());
+    if !simulate {
+        ceph_mon_command_without_data(cluster_handle, &cmd.to_string())?;
+    }
+    Ok(())
+}
+
+/// Possible values for the key are:
+/// full|pause|noup|nodown|noout|noin|nobackfill|norebalance|norecover|noscrub|
```

Comment:
Why not make the `key` arg into an Enum then? Something like

```rust
pub enum OsdOptionn {
  Full,
  Pause,
  ...,
}
```

And then you can derive Display for it to create the correct options

Reply:
I like it.